src/main/java/com/juniorstart/juniorstart/service/ProjectCommentService.java

```diff
@@ -0,0 +1,233 @@
+package com.juniorstart.juniorstart.service;
+
+import com.juniorstart.juniorstart.exception.BadRequestException;
+import com.juniorstart.juniorstart.exception.ResourceNotFoundException;
+import com.juniorstart.juniorstart.model.Project;
+import com.juniorstart.juniorstart.model.ProjectComment;
+import com.juniorstart.juniorstart.model.User;
+import com.juniorstart.juniorstart.payload.*;
+import com.juniorstart.juniorstart.repository.ProjectCommentRepository;
+import com.juniorstart.juniorstart.repository.ProjectLikeRepository;
+import com.juniorstart.juniorstart.repository.ProjectRepository;
+import com.juniorstart.juniorstart.repository.UserDao;
+import com.juniorstart.juniorstart.security.UserPrincipal;
+import com.juniorstart.juniorstart.util.ModelMapper;
+import com.juniorstart.juniorstart.util.ValidatePageUtil;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+
+
+/** Represents an comment service which manipulate project's comments data. 20-09-2020.
+ * @author Grzegorz Szczęsny
+ * @version 1.0
+ * @since 1.0
+ */
+@Service
+@Slf4j
+public class ProjectCommentService {
+
+    public ProjectCommentService(ProjectCommentRepository projectCommentRepository, UserDao userDao, ProjectRepository projectRepository, ProjectLikeRepository projectLikeRepository) {
+        this.projectCommentRepository = projectCommentRepository;
+        this.userDao = userDao;
+        this.projectRepository = projectRepository;
+        this.projectLikeRepository = projectLikeRepository;
+    }
+
+    final private ProjectCommentRepository projectCommentRepository;
+    final private UserDao userDao;
+    final private ProjectRepository projectRepository;
+    final private ProjectLikeRepository projectLikeRepository;
+
+    /** Create a new comment for a project.
+     * @param projectId A project's identification number.
+     * @param currentUser The current user credentials.
+     * @param commentRequest The comment’s credentials.
+     * @return a created comment
+     */
+    public ProjectCommentResponse createComment(CommentRequest commentRequest, UserPrincipal currentUser, Long projectId) {
+
+        User currentLoggedUser = userDao.findById(currentUser.getId()).orElseThrow(() -> new ResourceNotFoundException("User", "userId", currentUser.getId()));
```

Comment:
Jak używasz jakiejś metody wielokrotnie w innych metodach to ją wrzucasz do nowej i wywołujesz. Bo tak to piszesz jedno i to samo po kilka razy

Reply:
dzięki za radę.